code/datums/action.dm

```diff
@@ -106,9 +106,8 @@
 		button.desc = desc
 
 		ApplyIcon(button)
-
-		// If the action isn't available, darken the button
-		if(!IsAvailable())
+		var/obj/effect/proc_holder/spell/S = target
+		if(istype(S) && S.cooldown_handler.should_draw_cooldown())
```

Comment:
Won't this break it for non spell actions?

Reply:
Alright, returned back this check, now its 
`istype(S) && S.cooldown_handler.should_draw_cooldown() || !IsAvailable()`
So that wont change non cooldown actions, but also will draw cooldown on those with handler